projects/auth0-angular/src/lib/functional.ts

```diff
@@ -0,0 +1,35 @@
+import { HttpEvent, HttpRequest } from '@angular/common/http';
+import { inject } from '@angular/core';
+import { ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { Observable } from 'rxjs';
+import { AuthGuard } from './auth.guard';
+import { AuthHttpInterceptor } from './auth.interceptor';
+
+/**
+ * Functional AuthGuard to ensure routes can only be accessed when authenticated.
+ *
+ * Note: Should only be used as of Angular 15
+ *
+ * @param route Contains the information about a route associated with a component loaded in an outlet at a particular moment in time.
+ * @param state Represents the state of the router at a moment in time.
+ * @returns An Observable<Boolean>, indicating if the route can be accessed or not
+ */
+export const authGuardFn = (
+  route: ActivatedRouteSnapshot,
+  state: RouterStateSnapshot
+) => inject(AuthGuard).canActivate(route, state);
```

Comment:
This only uses canActivate. However, `canActivate` and `canActivateChild` are identical implementation wise. So people can use authGuardFn for both scenarios.